File: lib/services/feedback_service.dart
import 'package:dio/dio.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'proxy_ai_service.dart';

/// 反馈上报（优先后端 API，失败静默）
class FeedbackService {
  final Dio _dio;

  FeedbackService({Dio? dio}) : _dio = dio ?? Dio();

  Future<bool> submit({
    required String type,
    String? sceneId,
    String? sceneName,
    String? tone,
    String? textPreview,
    List<String>? reasons,
    String? comment,
  }) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      final base = prefs.getString('cleartalk_api_base_url')?.trim() ??
          ProxyAiService.defaultBaseUrl;

      await _dio.post(
        '$base/api/feedback',
        data: {
          'type': type,
          'sceneId': sceneId,
          'sceneName': sceneName,
          'tone': tone,
          'textPreview': textPreview,
          'reasons': reasons ?? [],
          'comment': comment ?? '',
          'meta': {'source': 'flutter'},
        },
        options: Options(
          headers: {'Content-Type': 'application/json'},
          sendTimeout: const Duration(seconds: 10),
          receiveTimeout: const Duration(seconds: 10),
        ),
      );
      return true;
    } catch (_) {
      return false;
    }
  }
}

File: src/services/feedback-service.js
/**
 * 用户反馈服务 — 本地缓存 + 后端同步
 */

import { api } from './api-client.js';
import { Storage } from '../core/storage.js';
import { logger } from '../core/logger.js';

export const FeedbackService = {
  /**
   * 提交反馈（优先后端，失败则仅本地）
   */
  async submit(payload) {
    const record = {
      ...payload,
      id: Date.now().toString(),
      createdAt: new Date().toISOString()
    };

    Storage.saveFeedback(record);

    try {
      const result = await api.post('/api/feedback', {
        type: payload.type,
        sceneId: payload.sceneId,
        sceneName: payload.sceneName,
        tone: payload.tone,
        reasons: payload.reasons,
        comment: payload.comment,
        textPreview: payload.textPreview,
        meta: {
          source: 'web',
          language: payload.language,
          userAgent: typeof navigator !== 'undefined' ? navigator.userAgent.slice(0, 120) : ''
        }
      });
      return { local: true, remote: true, id: result?.id };
    } catch (err) {
      logger.warn('Feedback API failed, saved locally only:', err.message);
      return { local: true, remote: false };
    }
  }
};

export default FeedbackService;
